app/src/test/java/com/github/se/eventradar/ScanTicketQrViewModelTest.kt

```diff
@@ -0,0 +1,171 @@
+  private lateinit var userRepository: IUserRepository
+  private lateinit var eventRepository: IEventRepository
+  private lateinit var qrCodeAnalyser: QrCodeAnalyser
+
+  private val myUID = "user1"
+
+  private val mockEvent1 =
+      Event(
+          eventName = "Event 1",
+          eventPhoto = "",
+          start = LocalDateTime.now(),
+          end = LocalDateTime.now(),
+          location = Location(0.0, 0.0, "Test Location"),
+          description = "Test Description",
+          ticket = EventTicket("Test Ticket", 0.0, 1),
+          mainOrganiser = "1",
+          organiserSet = mutableSetOf("Test Organiser"),
+          attendeeSet = mutableSetOf("user1", "user2", "user3"),
+          category = EventCategory.COMMUNITY,
+          fireBaseID = "1")
+
+  private val mockUser1 = // access to event 1 & 2 & 3
+      User(
+          userId = "user1",
+          birthDate = "01/01/2000",
+          email = "test@example.com",
+          firstName = "John",
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          accountStatus = "active",
+          eventsAttendeeSet = mutableSetOf("1", "2", "3"),
+          eventsHostSet = mutableSetOf("3"),
+          friendsSet = mutableSetOf(),
+          profilePicUrl = "http://example.com/pic.jpg",
+          qrCodeUrl = "http://example.com/qr.jpg",
+          username = "john_doe")
+
+  private val mockUser2 = // access to event 2 & 3 only
+      User(
+          userId = "user2",
+          birthDate = "01/01/2002",
+          email = "test@example2.com",
+          firstName = "John2",
+          lastName = "Doe2",
+          phoneNumber = "12345678902",
+          accountStatus = "active2",
+          eventsAttendeeSet = mutableSetOf("2", "3"),
+          eventsHostSet = mutableSetOf("32"),
+          friendsSet = mutableSetOf(),
+          profilePicUrl = "http://example.com/pic.jpg2",
+          qrCodeUrl = "http://example.com/qr.jpg2",
+          username = "john_doe2")
+
+  class MainDispatcherRule(
+      private val testDispatcher: TestDispatcher = UnconfinedTestDispatcher()
+  ) : TestWatcher() {
+    override fun starting(description: Description) {
+      Dispatchers.setMain(testDispatcher)
+    }
+
+    override fun finished(description: Description) {
+      Dispatchers.resetMain()
+    }
+  }
+
+  @get:Rule val mainDispatcherRule = MainDispatcherRule()
+
+  @Before
+  fun setUp() {
+    userRepository = MockUserRepository()
+    eventRepository = MockEventRepository()
+    qrCodeAnalyser = QrCodeAnalyser()
+    viewModel = ScanTicketQrViewModel(userRepository, eventRepository, qrCodeAnalyser)
+  }
+
+  @Test
+  fun testDecodingSuccess() = runTest {
+    userRepository.addUser(mockUser1)
+    eventRepository.addEvent(mockEvent1)
+    val testDecodedString = "user1"
+    qrCodeAnalyser.onTicketDecoded?.invoke(testDecodedString)
+    TestCase.assertEquals(testDecodedString, viewModel.uiState.value.decodedResult)
```

Comment:
nit: to stay consistent with the other test, I would just use `assertEquals` no needs for the `TestCase` in front of it